python/pyspark/sql/session.py

```diff
@@ -225,6 +230,9 @@ def __init__(self, sparkContext, jsparkSession=None):
         if SparkSession._instantiatedSession is None \
                 or SparkSession._instantiatedSession._sc._jsc is None:
             SparkSession._instantiatedSession = self
+            if self._jvm.SparkSession.getDefaultSession().isEmpty() \
+                    or not jsparkSession.equals(self._jvm.SparkSession.getDefaultSession().get()):
+                self._jvm.SparkSession.setDefaultSession(self._jsparkSession)
```

Comment:
Maybe we can simply overwrite the default session.

Reply:
@felixcheung has concern about simply overwriting the default session.

Reply:
I might miss something, but I guess @felixcheung's concern was fixed by checking if the default session is defined and not stopped so we can put the valid session or the same session from JVM without checking anymore.
But I'm okay to leave it as it is as well.